subprojects/code-quality/src/testFixtures/groovy/org/gradle/quality/integtest/fixtures/CodeNarcCoverage.groovy

```diff
@@ -18,14 +18,20 @@ package org.gradle.quality.integtest.fixtures
 
 import org.gradle.api.JavaVersion
 import org.gradle.api.plugins.quality.CodeNarcPlugin
+import org.gradle.util.TestPrecondition
 import org.gradle.util.VersionNumber
 
 class CodeNarcCoverage {
     private final static List<String> ALL = ["0.17", "0.21", "0.23", "0.24.1", "0.25.2", "1.0", "1.2.1", CodeNarcPlugin.DEFAULT_CODENARC_VERSION].asImmutable()
 
     private final static List<String> JDK11_SUPPORTED = ALL.findAll { VersionNumber.parse(it) > VersionNumber.parse("0.21") }.asImmutable()
+    // TODO: document this
+    private final static List<String> JDK14_SUPPORTED = ALL - ['0.17', '0.21', '1.0', '1.2.1', '1.5']
 
     static List<String> getSupportedVersionsByJdk() {
```

Comment:
Take a look at GroovyCoverage, I think it's simpler to think about which minimum version is required for whatever the feature is (JDK11 support, JDK14 support, etc) and filter the ALL list for that.

I was thinking of something like this:
```
class CodeNarcCoverage {
    private final static List<String> ALL = [CodeNarcPlugin.DEFAULT_CODENARC_VERSION, "0.17", "0.21", "0.23", "0.24.1", "0.25.2", "1.0", "1.2.1"].asImmutable()

    private final static VersionNumber JDK11_SUPPORTED = VersionNumber.parse("0.22")
    private final static VersionNumber JDK14_SUPPORTED = VersionNumber.parse("1.0")

    static List<String> getSupportedVersionsByJdk() {
        VersionNumber minimumVersion = VersionNumber.UNKNOWN
        if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_14)) {
            minimumVersion = JDK14_SUPPORTED
        } else if (JavaVersion.current().isJava11Compatible()) {
            minimumVersion = JDK11_SUPPORTED
        }
        return ALL.findAll {
            VersionNumber.parse(it) >= minimumVersion
        }
    }
}
```

This doesn't quite work yet because we're missing JavaVersion.VERSION_14